### ISSUE TITLE
Build fails (MSVC10)

### ISSUE THREAD

**Issue by AndreasMartin72** [utelle/wxsqlite3#8]
Hello Mr. Telle,

recent changes lead to a compile error:
```
1>ClCompile:
1>  wxsqlite3.cpp
1>..\src\wxsqlite3.cpp(4450): error C2511: 'int wxSQLite3FunctionContext::ExecAuthorizer(void *,int,const char *,const char *,const char *,const char *,const char *)' : overloaded member function not found in 'wxSQLite3FunctionContext'
1>          D:\git\wxsqlite3\include\wx/wxsqlite3.h(214) : see declaration of 'wxSQLite3FunctionContext'
1>
1>Build FAILED.
```
Reason:
wxsqlite3opt.h (included by wxsqlite3.cpp) defines WXSQLITE3_USER_AUTHENTICATION in line 31:
`#define WXSQLITE3_USER_AUTHENTICATION 1`
but wxsqlite3.h leaves WXSQLITE3_USER_AUTHENTICATION undefined. So declaration and definition of ExecAuthorizer() diverges...

Kindly regards,
Andreas Martin

**Comment by utelle** [comment]
Thanks for reporting.

I guess you used the MSVC solution files in  one of the directories `build30` or `build31`. These build files were not updated, and will be removed within the next days. Please use the solution files in directory **`build`**. They define the symbol `WXSQLITE3_USER_AUTHENTICATION` and therefore avoid the compilation problem.

However, that's no excuse. The problem could be solved by including `wxsqlite3opt.h` in `wxsqlite3.h` or by including `wxsqlite3opt.h` before `wxsqlite3.h` in `wxsqlite3.cpp`. But in fact, it is a design flaw to make the public header file `wxsqlite3.h` depending on the symbol `WXSQLITE3_USER_AUTHENTICATION`.

I will check how this dependency could possibly be removed.

**Comment by utelle** [comment]
Fixed in commit 3c51c3a09598a83092d404a28089572007e63fd9.

The dependency of `wxsqlite3.h`on the symbol `WXSQLITE3_USER_AUTHENTICATION` has been removed.

**Comment by AndreasMartin72** [comment]
Thanks a lot!

Kindly regards, 
Andreas Martin
